core/geometry/src/polyface/PolyfaceQuery.ts

```diff
@@ -273,6 +274,76 @@ export class PolyfaceQuery {
     return MomentData.inertiaProductsToPrincipalAxes(origin, inertiaProducts);
   }
 
+  /**
+   * Test for convex volume by dihedral angle tests on all edges.
+   *
+   * @param source mesh to examine
+   * * @param ignoreBoundaries if true, ignore simple boundary edges, i.e. allow unclosed meshes.
+   * @returns true if the mesh is closed and has all dihedral angles (angle across edge) positive
+   */
+  public static isConvexByDihedralAngleCount(source: Polyface, ignoreBoundaries: boolean = false): boolean {
+    return this.dihedralAngleSummary(source, ignoreBoundaries) > 0;
+  }
+
+  /**
+  * * Return a number summarizing the dihedral angles in the mesh.
+  * @param source mesh to examine
+  * @param ignoreBoundaries if true, ignore simple boundary edges, i.e. allow unclosed meshes.
+  *   * Return 1 if all angles are positive or planar.  The mesh is probably convex with outward normals.
+  *   * Return -1 if all angles are negative or planar.  The mesh is probably convex with inward normals.
+  *   * Return 0 if
+  *     * angles area mixed
+  *     * any edge has other than 1 incident facet or more than 2 incident facets.
+  *     * (but null edges are permitted -- These occur naturally at edges of quads at north or south pole)
+  */
+  public static dihedralAngleSummary(source: Polyface, ignoreBoundaries: boolean = false): number {
+    const edges = new IndexedEdgeMatcher();
+    const visitor = source.createVisitor(1);
+    visitor.reset();
+    const centroidNormal: Ray3d[] = [];
+    let normalCounter = 0;
+    while (visitor.moveToNextFacet()) {
+      const numEdges = visitor.pointCount - 1;
+      const normal = PolygonOps.centroidAreaNormal(visitor.point);
+      if (normal === undefined)
+        return 0;
+      centroidNormal.push(normal);
+      for (let i = 0; i < numEdges; i++) {
+        edges.addEdge(visitor.clientPointIndex(i), visitor.clientPointIndex(i + 1), normalCounter);
+      }
+      normalCounter++;
+    }
+    const badClusters: SortableEdgeCluster[] = [];
+    const manifoldClusters: SortableEdgeCluster[] = [];
+
+    edges.sortAndCollectClusters(manifoldClusters,
+      ignoreBoundaries ? undefined : badClusters, undefined, badClusters);
+    if (badClusters.length > 0)
+      return 0;
+    let numPositive = 0;
+    let numPlanar = 0;
+    let numNegative = 0;
+    const edgeVector = Vector3d.create();
+    for (const cluster of manifoldClusters) {
+      const sideA = cluster.at(0);
+      const sideB = cluster.at(1);
+      if (sideA instanceof SortableEdge
+        && sideB instanceof SortableEdge
+        && source.data.point.vectorIndexIndex(sideA.vertexIndexA, sideA.vertexIndexB, edgeVector)) {
+        const dihedralAngle = centroidNormal[sideA.facetIndex].direction.signedAngleTo(
+          centroidNormal[sideB.facetIndex].direction, edgeVector);
+        if (dihedralAngle.isAlmostZero) numPlanar++;
```

Comment:
`numPlanar` is computed but has no effect on the return value - should it?

Reply:
I added a return case for numPositive === numNegative === 0 and numPlanar nonzero.   All the dihedral angles are zero, so its a single plane -- call it convex.

BTW, if someone (a) sends in "ignoreBoundary" and (b) has multiple connected components that are all convex, the way the two interact could be nonconvex and would not be caught.